sqlglot/dialects/tsql.py

```diff
@@ -1067,3 +1123,21 @@ def drop_sql(self, expression: exp.Drop) -> str:
             if expression.args["kind"] == "VIEW":
                 expression.this.set("catalog", None)
             return super().drop_sql(expression)
+
+        def declare_sql(self, expression: exp.Declare) -> str:
+            declare_items = []
+            for declare_item in expression.expressions:
+                declare_item_sql = self.sql(declare_item)
+                declare_items.append(declare_item_sql)
+            declare_items_str = ", ".join(declare_items)
+
+            return f"DECLARE {declare_items_str}"
+
+        def declareitem_sql(self, expression: exp.DeclareItem) -> str:
+            variable = self.sql(expression, "this")
+            type = self.sql(expression, "kind")
+            # generating sql from a table looks like "TableName (Schema)",
+            # but since we already have the table name (as variable), replace the "name" with the type "TABLE"
+            type = type.replace(variable, "TABLE")
```

Comment:
I'm not sure how to avoid using this kind of hack here. I'm using the same method to parse the table info in `DECLARE @var AS TABLE (Schema)` as is used to parse `CREATE TABLE var (Schema)`. Therefore, the .`sql()` method naturally creates the same SQL for both statements. This is bad because the ordering of `TABLE` and `ExampleTableName` is reversed between the two statements. Without creating a different parsing method, specific to variable tables that generates a different Expression (which seems redundant and creates its own issues), I don't see how to avoid using this hack. Though, perhaps I'm missing some part of the codebase that's meant to handle this kind of thing in a cleaner way. Maybe I need a transform here?